kivy/tests/test_animations.py

```diff
@@ -35,11 +35,18 @@ def assert_(self, n_start, n_progress_greater_than_zero, n_complete):
     return EventCounter
 
 
+@pytest.fixture(scope='module')
+def approx():
+    from functools import partial
+    return partial(pytest.approx, abs=1)
+
+
 @pytest.fixture(autouse=True)
 def cleanup():
     from kivy.animation import Animation
-    for anim in Animation._instances.copy():
-        anim.cancel()
+    for anim in Animation._instances:
```

Comment:
It is concerning if `cancel` doesn't work. Shouldn't `cancel` work?

Reply:
This is a pure bug because `Animation.cancel()` takes one positional argument. The reason I didn't notice it at the time is `anim.cancel()` is excuted only when a test case fails, which didn't happen when I wrote it.

Reply:
Ok, but I think a better solution is to allow `cancel_all` https://github.com/kivy/kivy/blob/master/kivy/animation.py#L183 to take `None` for the `widget` parameter and then clear all the widgets. Because it's better to not access the private API outside `Animation`.